### PR TITLE
feat(cmd::api): work with go module

### LINKED ISSUE #270: `nirvana api` need to be compatible with `go mod`
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug

/kind feature

**What happened**:

I created a nirvana project with go mod and the project is not in `$GOPATH`. `nirvana api` in this project failed with messages:
```
INFO  0320-11:49:45.516+08 api.go:87 | No packages are specified, defaults to pkg/apis
WARN  0320-11:49:45.516+08 builder.go:69 | can't find project file, instead by default config
FATAL 0320-11:49:45.516+08 api.go:61 | can't find descriptors from []
```

**What you expected to happen**:

Generate api as normal.

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a nirvana project in `/tmp/myproject`
2. Run `nirvana api` in `/tmp/myproject`

**Anything else we need to know?**:

API generator can't find correct packages because of the mechanism of `go mod`. The generator should use system relative path rather than packages from `$GOPATH`.

### PR DESCRIPTION
**Pull request by knight42**
**What this PR does / why we need it**:

Make `nirvana api` work with go module.

Proof:
```
$ pwd
/Users/knight/github-caicloud/foo2
[20-10-26 14:31] [kind-kind] byted-mbp@ ~cchub/foo2
$ l
total 48K
drwxr-xr-x 2 knight staff   64 Oct 26 13:27 apis/
drwxr-xr-x 2 knight staff   64 Oct 26 13:27 bin/
drwxr-xr-x 3 knight staff   96 Oct 26 13:27 cmd/
drwxr-xr-x 3 knight staff   96 Oct 26 13:27 build/
drwxr-xr-x 3 knight staff   96 Oct 26 13:27 docs/
drwxr-xr-x 6 knight staff  192 Oct 26 13:27 hack/
drwxr-xr-x 8 knight staff  256 Oct 26 13:27 pkg/
drwxr-xr-x 3 knight staff   96 Oct 26 13:27 test/
-rw-r--r-- 1 knight staff  446 Oct 26 13:27 go.mod
-rw-r--r-- 1 knight staff  460 Oct 26 13:27 nirvana.yaml
-rw-r--r-- 1 knight staff  451 Oct 26 13:27 .golangci.yml
-rw-r--r-- 1 knight staff 5.7K Oct 26 13:27 Makefile
-rw-r--r-- 1 knight staff  544 Oct 26 13:27 README.md
-rw-r--r-- 1 knight staff  23K Oct 26 13:27 go.sum
drwxr-xr-x 7 knight staff  224 Oct 26 13:27 vendor/
[20-10-26 14:31] [kind-kind] byted-mbp@ ~cchub/foo2
$ nirvana api
INFO  1026-14:31:49.258+08 api.go:79 | No packages are specified, defaults to pkg/apis
INFO  1026-14:31:50.823+08 api.go:87 | Project root directory is /Users/knight/github-caicloud/foo2
INFO  1026-14:31:50.824+08 api.go:141 | Listening on 127.0.0.1:8080. Please open your browser to view api docs
```

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #270

Reference to #

**Special notes for your reviewer**:

/cc @kdada @iawia002 

**Release note**:

```release-note
cmd: `nirvana api` is compatible with go module now.
```

### CODE REVIEW

File: utils/project/path.go [hunk 1]
```diff
@@ -74,17 +74,37 @@ func PackageForPath(directory string) (string, error) {
...
+				}
+				_, absPath, err = GoPath(path)
+			}
+		}
 		if err != nil {
 			// Ignore the err and go next.
 			continue
 		}
-		err = filepath.Walk(absPath, func(path string, info os.FileInfo, err error) error {
+		_ = filepath.Walk(absPath, func(path string, info os.FileInfo, err error) error {
```
Comment (gaocegege) [inline utils/project/path.go:107]: Why don't we handle this error here?

Comment (knight42) [reply utils/project/path.go:107]: I am not sure, it is not handled now:
https://github.com/caicloud/nirvana/blob/2fb93f51598f353e72bbbd313a5ecb8d77388828/utils/project/path.go#L107

how should we handle this error properly?

Comment (gaocegege) [reply utils/project/path.go:107]: How about panic here?

Comment (iawia002) [reply utils/project/path.go:107]: I'm not so sure too, but I think we can keep ignoring it.

**Comment by iawia002** [comment]
I got an error in my local validation, seems a go version issue, what is the go version you are using?

```
error loading packages: go [list -e -json -compiled=true -test=false -export=false -deps=true -find=false --]: exit status 1: build .: cannot find module for path .
```

**Comment by knight42** [comment]
@iawia002 I am using go 1.15.3.
```
$ go version
go version go1.15.3 darwin/amd64
```

May i know the command that cause this error?

**Comment by iawia002** [comment]
> May i know the command that cause this error?

`packages.Load` returns this error, I tried on go 1.14.x and the issue just disappeared, seems that something wrong with go modules in 1.13.x

File: utils/project/path.go (hunk 1 above)
Comment (gaocegege) [reply utils/project/path.go:107]: OK SGTM

**Pull request merged**